src/Microsoft.Azure.SignalR.AspNet/ServerConnections/ServiceConnection.cs

```diff
@@ -213,8 +213,8 @@ private async Task OnConnectedAsyncCore(ClientConnectionContext clientContext, O
             var connectionId = message.ConnectionId;
             try
             {
-                clientContext.Transport =
-                    await _clientConnectionManager.CreateConnection(message, this);
+                clientContext.Transport = await _clientConnectionManager.CreateConnection(message);
+                clientContext.ServiceConnection = this;
```

Comment:
move the set of `ServiceConnection` to `.ctor` so that it is guaranteed to be not null at the very beginning?